pallets/baby-liminal/src/benchmarking/suite.rs

```diff
@@ -35,7 +40,14 @@ benchmarks! {
     overwrite_key {
         let l in 1 .. T::MaximumVerificationKeyLength::get();
         let key = vec![0u8; l as usize];
-    } : _(RawOrigin::Root, IDENTIFIER, key)
+        let _ = insert_key::<T>(key.clone ());
+    } : _(caller::<T>(), IDENTIFIER, key)
```

Comment:
I think this benchmark is incomplete. This benchmark does the following:

**[setup phase]**
1. Generate a key `k` of length `l`
2. Write `k` to storage

**[actual benchmark]**
1. `overwrite_key(k)`

Since `k` was already stored in the setup phase, the call to `overwrite_key` we're benchmarking will exit early when comparing old and new keys will have the same length (cmp operator would return `Equal`). We're skipping one `T::Currency::(un)reserve` and one write to storage (replacing previous deposit with new one.

Reply:
I believe the benchmark runs `overwrite` for keys of increasing length so the path with reserve is covered but unreserved is indeed missing.

Reply:
there is no diff in # read/writes whether it's reserve or unreserve.

Reply:
> I believe the benchmark runs `overwrite` for keys of increasing length so the path with reserve is covered but unreserved is indeed missing.

I would have thought that only `_(caller::<T>(), IDENTIFIER, key)` is benchmarked, not the setup phase, so either the setup is wrong and we keep overwriting there (`insert_key(k_len=1)`, `insert_key(key_len=2)`, ..., `insert_key(key_len=l)`) and benchmark just the last insert or the benchmarked method is run for every setup separately - we insert key with length 1 and try to overwrite with the same key etc.

Reply:
I can't believe that this:
```rust
Equal => {
                            // do nothing
                        }
```
is equal in terms of complexity to:
```rust
Greater => {
    // lock the difference deposit - prev
    let difference = deposit - previous_deposit;
    T::Currency::reserve(&who, difference)
        .map_err(|_| Error::<T>::CannotAffordDeposit)?;
    *maybe_previous_deposit = Some(deposit);
}
```
but it seems like I'm the only one interested in checking this so I'll approve it now.